File: ITapply/ITapply.Models/SearchObjects/ApplicationSearchObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static ITapply.Models.Responses.EnumResponse;

namespace ITapply.Models.SearchObjects
{
    public class ApplicationSearchObject : BaseSearchObject
    {
        public int? CandidateId { get; set; }
        public int? JobPostingId { get; set; }
        public int? EmployerId { get; set; }
        public string JobTitle { get; set; } = string.Empty;
        public string CandidateName { get; set; } = string.Empty;
        public string CompanyName { get; set; } = string.Empty;
        public ApplicationStatus? Status { get; set; }
        public DateTime? ApplicationDateFrom { get; set; }
        public DateTime? ApplicationDateTo { get; set; }
    }
} 

File: ITapply/ITapply.UI/itapply_mobile/lib/models/search_objects/application_search_object.dart
import 'package:itapply_mobile/models/enums.dart';
import 'package:itapply_mobile/models/search_objects/base_search_object.dart';
import 'package:json_annotation/json_annotation.dart';

part 'application_search_object.g.dart';

@JsonSerializable()
class ApplicationSearchObject extends BaseSearchObject {
  int? CandidateId;
  int? JobPostingId;
  int? EmployerId;
  String? JobTitle;
  String? CandidateName;
  String? CompanyName;
  @JsonKey(toJson: _applicationStatusToJson)
  ApplicationStatus? Status;
  DateTime? ApplicationDateFrom;
  DateTime? ApplicationDateTo;

  ApplicationSearchObject({
    this.CandidateId,
    this.JobPostingId,
    this.EmployerId,
    this.JobTitle,
    this.CandidateName,
    this.CompanyName,
    this.Status,
    this.ApplicationDateFrom,
    this.ApplicationDateTo,
    super.Page,
    super.PageSize,
    super.IncludeTotalCount,
    super.RetrieveAll,
  });

  @override
  Map<String, dynamic> toJson() => _$ApplicationSearchObjectToJson(this);
}

int? _applicationStatusToJson(ApplicationStatus? status) => status?.index;
